File: MonkeyGameJava/app/src/main/java/com/example/monkeygamejava/GameActivity.java
package com.example.monkeygamejava;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.text.BreakIterator;

public class GameActivity extends AppCompatActivity implements View.OnTouchListener{
    public static boolean isLeftPressed = false; // нажата левая кнопка
    public static boolean isRightPressed = false; // нажата правая кнопка
    public static TextView textViewScore;
    public static TextView textViewResult;
    public static int result=0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_game);
        GameView gameView = new GameView(this); // создаём gameView
        LinearLayout gameLayout = (LinearLayout) findViewById(R.id.gameLayout); // находим gameLayout
        gameLayout.addView(gameView); // и добавляем в него gameView
        Button leftButton = (Button) findViewById(R.id.leftButton); // находим кнопки
        Button rightButton = (Button) findViewById(R.id.rightButton);
        textViewScore = findViewById(R.id.textViewScore);
        textViewResult=findViewById(R.id.textViewResult);
        leftButton.setOnTouchListener(this); // и добавляем этот класс как слушателя (при нажатии сработает onTouch)
        rightButton.setOnTouchListener(this);
    }

    public static void setScore(int score){
        result=result+score;
        textViewScore.setText(""+result);
    }

    public static void showScore(){
        textViewResult.setText("Your result:"+result);
    }

    public boolean onTouch(View button, MotionEvent motion) {
        switch(button.getId()) { // определяем какая кнопка
            case R.id.leftButton:
                switch (motion.getAction()) { // определяем нажата или отпущена
                    case MotionEvent.ACTION_DOWN:
                        isLeftPressed = true;
                        break;
                    case MotionEvent.ACTION_UP:
                        isLeftPressed = false;
                        break;
                }
                break;
            case R.id.rightButton:
                switch (motion.getAction()) { // определяем нажата или отпущена
                    case MotionEvent.ACTION_DOWN:
                        isRightPressed = true;
                        break;
                    case MotionEvent.ACTION_UP:
                        isRightPressed = false;
                        break;
                }
                break;
        }
        return true;
    }
}

File: MonkeyGame/app/src/main/java/com/example/monkeygamejava/GameActivity.kt
package com.example.monkeygamejava

import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.view.View.OnTouchListener
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class GameActivity : AppCompatActivity(), OnTouchListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_game)
        val gameView = GameView(this, object : OnDieListener {
            override fun onDie() {
                this@GameActivity.runOnUiThread {
                    this@GameActivity.finish()
                    Toast.makeText(baseContext,"Your result :$result", Toast.LENGTH_SHORT).show()
                    result=0
                }


            }
        })
        gameLayout = findViewById<View>(R.id.gameLayout) as LinearLayout
        gameLayout!!.addView(gameView)
        val leftButton = findViewById<View>(R.id.leftButton) as Button
        val rightButton = findViewById<View>(R.id.rightButton) as Button
        textViewScore = findViewById(R.id.textViewScore)

        leftButton.setOnTouchListener(this)
        rightButton.setOnTouchListener(this)
    }


    override fun onTouch(button: View, motion: MotionEvent): Boolean {
        when (button.id) {
            R.id.leftButton -> when (motion.action) {
                MotionEvent.ACTION_DOWN -> isLeftPressed = true
                MotionEvent.ACTION_UP -> isLeftPressed = false
            }
            R.id.rightButton -> when (motion.action) {
                MotionEvent.ACTION_DOWN -> isRightPressed = true
                MotionEvent.ACTION_UP -> isRightPressed = false
            }
        }
        return true
    }

    companion object {
        var isLeftPressed = false
        var isRightPressed = false
        var textViewScore: TextView? = null
        var gameLayout: LinearLayout? = null
        var result = 0
        fun setScore(score: Int) {
            result += score
            textViewScore!!.text = "Your score: $result"
        }
    }
}